core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -869,7 +876,19 @@ private[spark] class BlockManager(
           blockInfoManager.unlock(blockId)
         }
       } else {
-        blockInfoManager.removeBlock(blockId)
+        // If an exception was thrown then it's possible that the code in `putBody` has already
+        // notified the master about the availability of this block, so we need to send an update
+        // to remove this block location.
+        removeBlockInternal(
+          blockId, tellMaster = tellMaster && putBlockInfo.tellMaster && exceptionWasThrown)
```

Comment:
aren't these two tellMaster values equivalent since we set it in the block from the function arg above?